src/media/media.service.ts

```diff
@@ -1,32 +1,77 @@
 import { Injectable } from '@nestjs/common';
 import { AuthUser } from 'src/common/decorators';
 import { PrismaService } from 'src/prisma/prisma.service';
-import { RestaurantService } from 'src/restaurant/restaurant.service';
 import { S3Service } from 'src/common/services/s3/s3.service';
 import { v4 as uuidv4 } from 'uuid';
 import { SharpService } from 'src/common/services/sharp';
+import { Prisma } from '@prisma/client';
 
 @Injectable()
 export class MediaService {
+  private ORIGINAL_IMAGE_RESOLUTION = 1024;
+
+  private MEDIUM_THUMBNAIL_IMAGE_RESOLUTION = 256;
+
+  private SMALL_THUMBNAIL_IMAGE_RESOLUTION = 128;
+
   constructor(
     private prisma: PrismaService,
-    private restaurantService: RestaurantService,
     private s3Service: S3Service,
     private sharpService: SharpService,
   ) {}
 
-  async uploadMenuItemImage(image: Express.Multer.File, currentUser: AuthUser) {
-    const restaurant =
-      await this.restaurantService.getRestaurantByOwnerId(currentUser);
-    const formatImage = await this.sharpService.formatImage(image);
-    const imageSlug = `restaurants/${restaurant.id}/menu/${uuidv4()}.webp`;
-    await this.s3Service.uploadFile(imageSlug, formatImage);
+  async uploadMenuItemImage(image: Express.Multer.File, user: AuthUser) {
+    const originalImage = await this.sharpService.resize(
+      image,
+      this.ORIGINAL_IMAGE_RESOLUTION,
+    );
+
+    const mediumThumbnailImage = await this.sharpService.resize(
+      image,
+      this.MEDIUM_THUMBNAIL_IMAGE_RESOLUTION,
+    );
+
+    const smallThumbnailImage = await this.sharpService.resize(
+      image,
+      this.SMALL_THUMBNAIL_IMAGE_RESOLUTION,
+    );
+
+    const originalImageSlug = `restaurants/${user.restaurantId}/menu/${uuidv4()}.webp`;
```

Comment:
a lot of code repeating. refactor.

you could have single method that takes resolution, resizes and uploads to s3